docs/389ds/design/audit-json-logging-design.md

```diff
@@ -0,0 +1,74 @@
+---
+title: "Audit JSON Logging Design"
+---
+
+# Audit JSON Logging
+----------------
+
+Overview
+--------
+
+In order to make the audit logging more consumable by standard parsing tool it will written in a JSON format instead of using a partial LDIF format which not LDAP compliant in its current form.
+
+While the old format, after a bit if data massaging, could be replayed using a LDAP client (e.g. ldapmodify), this is most likely not used by most customers.  Instead, customers are looking for actual audit information that can by consumed by well known log parsers.
+
+JSON Design
+------------
+
+```
+[
+    {
+        date: 20243457458475847Z
+        target_dn: DN,
+        bind_dn: DN,
+        client: IP_ADDRESS,
+        conn_id: ####,
+        op_id: ####,
+        result: ##,
+        changetype: add/delete/modify/modrdn,
+        add: {
+            "attr1": [value, value, ...],
+            "attr2": [value, value, ...],
+        },
+        delete: DN,
+        modify: [
+           {
+             op: add/replace/delete,
+             attr: "cn",
+             delete_value: "???",  # Delete operation only (remove specific value from multi-valued attribute)
+             new_value: "name",
+           },
+        ],
+        modrdn: {
+            deleteOldRdn: True/False,
+            newRdn: "cn=mark",
+        }
+    },
+    {
+        ...
+    }
+]
+```
+
+Configuration
+------------------------
+
+Add a new configuration setting for audit logging under **cn=config**
+
+```
+nsslapd-auditlog-json-format: on/off
```

Comment:
When the switch is flipped, will the server log to the same file as with the old format? Will it rotate the log to avoid mixing 2 formats? Should it use a different filename with `.json` extension?

Reply:
I was going to keep it the same file name, but yes it should trigger a rotation

Reply:
Thanks for the confirmation.